backend/src/controllers/CMSTutorialController.ts

```diff
@@ -1,33 +1,33 @@
 import { Request, Response } from "express";
 import { ObjectId, MongoError } from "mongodb";
-import { Subject } from "rxjs";
 import jwt from "jsonwebtoken";
 import Tutorial from "../dtos/Tutorial.dto";
 import PartialTutorial from "../dtos/PartialTutorial.dto";
 import { INTERNAL_ERROR_TEXT, SUCCESSFUL_DOC_CREATION, BAD_REQUEST_TEXT } from "../constants";
-import TutorialDbService from "../services/TutorialDbService";
+import TutorialUpdateService from "../services/TutorialUpdateService";
 
 /**
  * Tutorial route controller
  *
  * @author ale8k
  */
-export default class TutorialController {
+export default class CMSTutorialController {
     /**
      * HTTP method handlers
      */
-    public static get: (req: Request, res: Response) => void = TutorialController.getAllTutorials;
-    public static post: (req: Request, res: Response) => void = TutorialController.createTutorial;
-    public static put: (req: Request, res: Response) => void = TutorialController.updateTutorialById;
+    public static get: (req: Request, res: Response) => void = CMSTutorialController.getAllTutorials;
+    public static post: (req: Request, res: Response) => void = CMSTutorialController.createTutorial;
+    public static put: (req: Request, res: Response) => void = CMSTutorialController.updateTutorialById;
 
     /**
      * Grabs all the {@link tutorialDocs Tutoriall[]} and responds
      *
+     * @async
      * @param {Request} req the users request obj
      * @param {Response} res our res obj
      */
     private static getAllTutorials(req: Request, res: Response): void {
-        TutorialDbService.getAllDocuments<Tutorial>("tutorials").then(
+        TutorialUpdateService.getAllDocuments<Tutorial>("tutorials").then(
             (tutorials) => res.status(200).json(tutorials).end(),
             (err) => res.status(500).send(INTERNAL_ERROR_TEXT + JSON.stringify(err)).end()
         );
```

Comment:
Can you make use of `async await` to handle this promises?

Reply:
Some mongo methods don't return anything from the promises so nope, we can't. For some, we can, but then it's inconsistent for the sake of putting generators in (async/await)